src/charm.py

```diff
@@ -510,19 +502,22 @@ def _configure(self):  # noqa: C901
             tsdb_versions_migration_dates=self._tsdb_versions_migration_dates,
         ).build()
 
-        restart = self._update_config(config) or restart
-        self._loki_container.add_layer(self._name, new_layer, combine=True)
-        # Now that we for sure have a layer, we can check if the service is running
-        restart = not self._loki_container.get_service(self._service_name).is_running() or restart
+        has_config_changed = self._update_config(config)
+        self._loki_container.add_layer(self._name, self._loki_pebble_layer, combine=True)
```

Comment:
Still a bit unclear to me what the purpose is of adding a layer which potentially is identical to the one we have already. Can you add a comment to explain what's going on?

Reply:
if the layer is unchanged, do we still want to restart?

Reply:
Technically the pebble layer will always be the same because the command will always be:
`"command": "/usr/bin/loki -config.expand-env=true -config.file=LOKI_CONFIG_DIR/loki-local-config.yaml"`
and the rest of the layer are static strings. I could add code to check `if previous_layer != current_layer` then `replan`.

Reply:
The main reason we add a layer here without checking for layer differences is due to the comment you see on line `515` "# Now that we for sure have a layer, we can check if the service is running". Which we check with `container_is_running()` a few lines later in the code.

Is there a better way to handle ensuring that we can check if the container is running without adding a layer?

Reply:
adding a layer has, as far as I know, no effect on the running/restart behaviour unless the plan changes as a result.
To check the service is running I would `self._loki_container.get_service(self._service_name).is_running()` as it was before, you shouldn't need to re-add the layer before.
Not saying it won't work, I just think it's effectively a no-op